numba/parfor.py

```diff
@@ -1009,6 +1022,83 @@ def find_mask_from_size(size_var):
                     for l in body_labels:
                         blocks.pop(l)
 
+    def _replace_loop_access_indices(self, loop_body, index_set, new_index):
+        """
+        Replace array access indices in a loop body with a new index.
+        index_set has all the variables that are equivalent to loop index.
+        """
+        # treat new index like others since replacing it with itself is ok
+        index_set.add(new_index.name)
+
+        # add dummy return to last block for CFG computation
+        # max is last block since we add it manually for prange
+        last_label = max(loop_body.keys())
+        loop_body[last_label].body.append(
```

Comment:
Maybe we should consider doing this add/remove dummy return in Python "with" style.

Reply:
Good idea. Done.